### PR TITLE
[jellyfin] add play by id channels and update sdk

### PR DESCRIPTION
**Pull request by GiviMAD**
Signed-off-by: Miguel Álvarez <miguelwork92@gmail.com>
Add channels for play media by item id and update sdk deps.

### CODE REVIEW

File: bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java [hunk 1]
```diff
@@ -39,10 +45,12 @@
 import org.eclipse.jdt.annotation.Nullable;
 import org.jellyfin.sdk.api.client.exception.ApiClientException;
 import org.jellyfin.sdk.model.api.BaseItemDto;
+import org.jellyfin.sdk.model.api.BaseItemKind;
 import org.jellyfin.sdk.model.api.PlayCommand;
 import org.jellyfin.sdk.model.api.PlayerStateInfo;
 import org.jellyfin.sdk.model.api.PlaystateCommand;
 import org.jellyfin.sdk.model.api.SessionInfo;
+import org.jetbrains.annotations.NotNull;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java:54]: Strange import, to be checked.
Your intention was probably rather to use JDT annotation.

File: bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java [hunk 2]
```diff
@@ -183,6 +215,14 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
+    @NotNull
```
Comment (lolodomo) [inline bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java:220]: Related to my other comment

File: bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java [hunk 3]
```diff
@@ -183,6 +215,14 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
+    private UUID parseItemUUID(Command command) {
+        var itemId = command.toFullString().replace("-", "");
+        UUID itemUUID = new UUID(new BigInteger(itemId.substring(0, 16), 16).longValue(),
+                new BigInteger(itemId.substring(16), 16).longValue());
```
Comment (lolodomo) [inline bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java:224]: I guess some exception handling should be done (NumberFormatException).

File: bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java [hunk 4]
```diff
@@ -423,6 +468,20 @@ private void playItemInternal(BaseItemDto item, PlayCommand playCommand, @Nullab
         getServerHandler().playItem(lastSessionId, playCommand, item.getId().toString(), startPositionTicks);
     }
 
+    private void runItemById(UUID itemId, @Nullable PlayCommand playCommand)
+            throws SyncCallback.SyncCallbackError, ApiClientException {
+        var item = getServerHandler().getItem(itemId, null);
+        if (item == null) {
+            logger.error("Unable to find item with id: {}", itemId);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java:485]: Please reduce log level to DEBUG or WARN

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by GiviMAD** [comment]
I have added a channel to read the id of the item currently playing.
No more changes, sorry for the late addition.

**Comment by lolodomo** [comment]
> I have added a channel to read the id of the item currently playing.
> No more changes, sorry for the late addition.

Not yet reviewed, you publish while I was reviewing the previous code.

**Comment by GiviMAD** [comment]
> Not yet reviewed, you publish while I was reviewing the previous code.

Yes, sorry, I should have written in the pr that I have missed that one. 
No more changes to be done, I will start with the review you already submitted.

File: bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java (hunk 2 above)
Comment (GiviMAD) [reply bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java:220]: Seems like I don't need it at all, I think it got added when I generated the method with the refactor into method functionality of the IDE. Thanks for pointing it.

File: bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java [hunk 5]
```diff
@@ -141,6 +149,55 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
                     break;
+                case PLAY_BY_ID_CHANNEL:
+                    if (command instanceof RefreshType) {
+                        return;
+                    }
+                    UUID itemUUID;
+                    try {
+                        itemUUID = parseItemUUID(command);
+                    } catch (NumberFormatException e) {
+                        logger.warn("Thing {}: Unable to parse item UUID.", thing.getUID());
```
Comment (lolodomo) [inline bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java:160]: Better option:
```
                         logger.warn("Thing {}: Unable to parse item UUID in command {}.", thing.getUID(), command);
```
Same for others.

File: bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java [hunk 6]
```diff
@@ -517,10 +609,11 @@ private void seekToTick(long seekPositionTick) throws SyncCallback.SyncCallbackE
     }
 
     private void cleanChannels() {
-        List.of(MEDIA_CONTROL_CHANNEL, PLAYING_ITEM_PERCENTAGE_CHANNEL, PLAYING_ITEM_NAME_CHANNEL,
-                PLAYING_ITEM_SERIES_NAME_CHANNEL, PLAYING_ITEM_SEASON_NAME_CHANNEL, PLAYING_ITEM_SEASON_CHANNEL,
-                PLAYING_ITEM_EPISODE_CHANNEL, PLAYING_ITEM_GENRES_CHANNEL, PLAYING_ITEM_TYPE_CHANNEL,
-                PLAYING_ITEM_SECOND_CHANNEL, PLAYING_ITEM_TOTAL_SECOND_CHANNEL).forEach(this::cleanChannel);
+        List.of(MEDIA_CONTROL_CHANNEL, PLAYING_ITEM_PERCENTAGE_CHANNEL, PLAYING_ITEM_ID_CHANNEL,
```
Comment (lolodomo) [inline bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java:612]: No need to add the four other channels you added ?

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java (hunk 6 above)
Comment (GiviMAD) [reply bundles/org.openhab.binding.jellyfin/src/main/java/org/openhab/binding/jellyfin/internal/handler/JellyfinClientHandler.java:612]: No, I'm only clearing the channels related to the playback state.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**